source/develop/release-management/features/sla/hosted-engine-network-check.html.md

```diff
@@ -0,0 +1,115 @@
+---
+title: Alternative Network Check for Self Hosted Engine
+category: feature
+authors: dholler
+feature_name: Alternative Network Check for Self Hosted Engine
+feature_modules: ovirt-hosted-engine-ha, cockpit-ovirt, ovirt-ansible-hosted-engine-setup, ovirt-hosted-engine-setup
+feature_status: Released
+---
+
+# Alternative Network Check for Self Hosted Engine
+
+## Summary
+
+Currently ovirt-hosted-engine-ha, the agent running on potential hosts to host
+the Self Hosted Engine VM, checks his network liveliness by pinging the
+gateway.
+This feature adds alternative ways to check network liveliness, to enable
+Self Hosted Engine in networks which does not allow pinging the gateway.
+
+The progress of the implementation and testing of this feature is tracked on
+[Bug 1659052 - [RFE] Consider alternative to ICMP ping for hosted engine network test](https://bugzilla.redhat.com/1659052)
+
+## Owner
+
+*   Name: [Dominik Holler](https://github.com/dominikholler)
+
+*   Email: <dholler@redhat.com>
+
+
+## Detailed Description
+
+Additional ways to check the network liveliness to be added are DNS, TCP and
+none.
+
+DNS means that an NS query for "." (the root) will be performed, if this
+succeeds, network is considered as alive.
+
+TCP means, that a TCP connection to a given host and port combination will be
+created. If this succeeds, network is considered as alive.
+
+None means, that the network is always considered alive.
+
+This feature requires that on deployment of the self hosted engine up to three
+additional parameters:
+
+- The desired way of checking network liveliness.
+- The destination host for the TCP liveliness check.
+- The destination port for the TCP liveliness check.
+
+## Benefit to oVirt
+
+This feature will enable the usage of Self Hosted Engine in networks which does
+not allow pinging the gateway.
+
+## Modification of ovirt-hosted-engine-setup
+
+`hosted-engine --deploy` is going to be extended to ask and check the additional
+parameters like this:
+
+~~~~
+          Please specify which way the network connectivityshould be checked (ping, dns, tcp, none) [dns]: tcp
+          Please specify the desired destination IP addressof the TCP connection test: sdf
+          Please specify the desired destination TCP portof the TCP connection test: 434
+[ ERROR ] Failed to connect via TCP
+~~~~
+
+The related variables in the answer file will be
+`OVEHOSTED_NETWORK/network_test`, `OVEHOSTED_NETWORK/network_test_tcp_address`
+and `OVEHOSTED_NETWORK/network_test_tcp_port`.
```

Comment:
is port mandatory? What's the value in case of ICMP?

Reply:
I added "while the latter two are only relevant if the value of OVEHOSTED_NETWORK/network_test is tcp."

Reply:
> is port mandatory? What's the value in case of ICMP?
ICMP is not an application, its a layer 3 protocol, it's protocol number is 1 and its acting at network layer, hence it doesn't has any port. Ports are for transport layer 4.